pkg/service/catalog/api/types.go

```diff
@@ -0,0 +1,21 @@
+package api
+
+import (
+	kapi "k8s.io/kubernetes/pkg/api"
+)
+
+type CatalogService struct {
+	// Target is a reference to a service
+	Target kapi.ObjectReference
+	// Type is a published service's claim type
+	Type string
+	// Description is a user-defined description of the published service
+	Description string
```

Comment:
My intent with having a description in CatalogService is to use it as a materialized view (read: copy) of the original Service's description. Users can't create CatalogServices. They are realized by virtue of a Service having an annotation indicating that it is for publication. It still seems much clearer to me to have a first class field on CatalogService to represent description.